Dockerfile

```diff
@@ -933,6 +933,31 @@ COPY --from=sbom-amd64 / /
 
 FROM sbom-container-${TARGETARCH} AS sbom-container-target
 
+# Use an unpinned latest version, because we want to use the latest advisories
+FROM ghcr.io/siderolabs/generate-vex:latest AS talos-vex
+
+FROM build-go AS vex-generate
+ARG TAG
+COPY --from=talos-vex /generate-vex /generate-vex
+# We do not need reproducible timestamps here, as we only use VEX for scanning
+# FIXME: add an output option to not require a shell in this step
+RUN /generate-vex gen --target-version $TAG > /talos.vex.json
+
+FROM scratch AS vex
+COPY --from=vex-generate /talos.vex.json /talos.vex.json
+
+FROM build-go AS grype-scan
+COPY --from=sbom-arm64 /talos-arm64.spdx.json /talos-arm64.spdx.json
+COPY --from=vex /talos.vex.json /talos.vex.json
+COPY ./tools ./tools
+# TODO: should we cache DB, how often is it updated?
+RUN --mount=type=cache,target=/.cache,id=talos/.cache go tool -modfile=tools/go.mod \
+    github.com/anchore/grype/cmd/grype sbom:/talos-arm64.spdx.json \
+    --vex /talos.vex.json -vv --fail-on negligible 2>&1 | tee /grype-scan.log
```

Comment:
Here we need to both fail and produce a readable log file

Reply:
make this a local target which passes, and we can parse the output file to see if we need to fail